infrastructure/src/test/java/org/corfudb/infrastructure/log/statetransfer/batchprocessor/protocolbatchprocessor/ProtocolBatchProcessorTest.java

```diff
@@ -0,0 +1,246 @@
+
+        ReadBatch join = res.join();
+        assertThat(join.getStatus() == ReadBatch.ReadStatus.SUCCEEDED).isTrue();
+        assertThat(join.getData()).isEqualTo(secondReturnedRecords);
+
+    }
+
+    /**
+     * Handle time outs..
+     */
+    @Test
+    void retryReadRecordsTimeout() {
+        List<Long> addresses = LongStream.range(0L, 10L).boxed().collect(Collectors.toList());
+
+        List<LogData> secondReadList = createStubList(addresses);
+
+        ReadOptions readOptions = ReadOptions.builder()
+                .waitForHole(true)
+                .clientCacheable(false)
+                .serverCacheable(false)
+                .build();
+        LogUnitClient logUnitClient = mock(LogUnitClient.class);
+
+        doReturn(CompletableFuture.completedFuture(true)).when(logUnitClient).writeRange(secondReadList);
+
+        AddressSpaceView addressSpaceView = mock(AddressSpaceView.class);
+
+        doAnswer(answer -> {
+            throw new TimeoutException();
+        }).when(addressSpaceView).simpleProtocolRead(addresses, readOptions);
+
+        ProtocolBatchProcessor batchProcessor = ProtocolBatchProcessor
+                .builder()
+                .logUnitClient(logUnitClient)
+                .addressSpaceView(addressSpaceView)
+                .build();
+        ProtocolBatchProcessor spy = spy(batchProcessor);
+
+        CompletableFuture<ReadBatch> res =
+                spy.retryReadRecords(TransferBatchRequest.builder().addresses(addresses).build(), 0);
+
+        assertThat(res).isCompletedExceptionally();
+
+    }
+
+
+    @Test
+    void checkReadRecordsComplete() {
+        List<Long> addresses = LongStream.range(0L, 10L).boxed().collect(Collectors.toList());
+        List<LogData> stubList = createStubList(addresses);
+        Map<Long, ILogData> stubMap = createStubMap(stubList);
+        ReadOptions readOptions = ReadOptions.builder()
+                .waitForHole(true)
+                .clientCacheable(false)
+                .serverCacheable(false)
+                .build();
+        LogUnitClient logUnitClient = mock(LogUnitClient.class);
+        doReturn(CompletableFuture.completedFuture(true)).when(logUnitClient).writeRange(stubList);
+        AddressSpaceView addressSpaceView = mock(AddressSpaceView.class);
+        doReturn(stubMap).when(addressSpaceView).simpleProtocolRead(addresses, readOptions);
+        ProtocolBatchProcessor batchProcessor = ProtocolBatchProcessor
+                .builder()
+                .logUnitClient(logUnitClient)
+                .addressSpaceView(addressSpaceView)
+                .build();
+        ReadBatch res = batchProcessor.checkReadRecords(addresses, stubMap, Optional.empty());
+        List<LogData> expected = getRecordsFromStubMap(stubMap);
+        assertThat(res.getStatus() == ReadBatch.ReadStatus.SUCCEEDED).isTrue();
+        assertThat(res.getData()).isEqualTo(expected);
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651453)